src/jit/codegenarm.cpp

```diff
@@ -44,7 +44,39 @@ void CodeGen::genSetRegToIcon(regNumber reg, ssize_t val, var_types type, insFla
 //
 void CodeGen::genEmitGSCookieCheck(bool pushReg)
 {
-    NYI("ARM genEmitGSCookieCheck");
+    noway_assert(compiler->gsGlobalSecurityCookieAddr || compiler->gsGlobalSecurityCookieVal);
+
```

Comment:
You right, code is mostly the same, I've moved it to the codegencommon. But I had to guard with few `#ifdef` statements which looks in my opinion little bit ugly.

Furthermore a lot of code for both ARM platforms is identical. It looks there is need to make some abstraction level (e.g. codegenarmcommon) for this platforms. Do you have ideas how to do it?